File: LeetCode/Problem189_rotate_array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    if (k === 0 || nums.length <= 1) return;
    k %= nums.length;
    head = nums.slice(nums.length - k);
    tail = nums.slice(0, nums.length - k);
    nums.length = 0;
    nums.push(...[...head, ...tail]);
};

File: LeetCode/Problem189_rotate_array.java
public class Solution {
    public void rotate(int[] nums, int k) {
        if (k == 0 || nums.length <= 1) return;
        k %= nums.length;
        int[] temp = new int[k];
        System.arraycopy(nums, nums.length - k, temp, 0, k);
        System.arraycopy(nums, 0, nums, k, nums.length - k);
        System.arraycopy(temp, 0, nums, 0, temp.length);
    }
}
